topics.go

```diff
@@ -0,0 +1,83 @@
+package sdk
+
+import (
+	"context"
+	"errors"
+
+	"github.com/oklog/ulid/v2"
+	api "github.com/rotationalio/go-ensign/api/v1beta1"
+)
+
+const DefaultPageSize uint32 = 100
+
+// Check if a topic with the specified name exists in the project or not.
+func (c *Client) TopicExists(ctx context.Context, topicName string) (_ bool, err error) {
+	// TODO: more directly use an exists RPC call for faster checking with an index.
+	var topics []*api.Topic
+	if topics, err = c.ListTopics(ctx); err != nil {
+		return false, err
+	}
+
+	for _, topic := range topics {
+		if topic.Name == topicName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// Create topic with the specified name and return the topic ID if there was no error.
+func (c *Client) CreateTopic(ctx context.Context, topic string) (_ string, err error) {
+	var reply *api.Topic
+	if reply, err = c.api.CreateTopic(ctx, &api.Topic{Name: topic}); err != nil {
```

Comment:
Here is where `CreateTopic` got moved to, but `api` is not exposed to the user! Whoops!